edge-util/src/Microsoft.Azure.Devices.Edge.Util/metrics/appMetrics/MetricsTimer.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Util.Metrics.AppMetrics
+{
+    using System;
+    using System.Collections.Generic;
+    using App.Metrics;
+    using App.Metrics.Timer;
+
+    public class MetricsTimer : BaseMetric, IMetricsTimer
+    {
+        readonly IMeasureTimerMetrics timerMetrics;
+        readonly TimerOptions timerOptions;
+
+        public MetricsTimer(string name, IMeasureTimerMetrics timerMetrics, List<string> labelNames)
+            : base(labelNames, new List<string>())
+        {
+            this.timerMetrics = timerMetrics;
+            this.timerOptions = new TimerOptions
+            {
+                Name = name,
+                MeasurementUnit = Unit.Items,
+                DurationUnit = TimeUnit.Milliseconds,
+                RateUnit = TimeUnit.Milliseconds
```

Comment:
Does this support floats and then `seconds` for a unit?

Reply:
It does.. but the trouble is the histogram (which is used as a timer), which only supports longs.